Assets/HoloToolkit/Input/Scripts/Utilities/AttachToController.cs

```diff
@@ -42,90 +40,56 @@ public class AttachToController : MonoBehaviour
 
         public bool IsAttached { get; private set; }
 
-        private Transform elementTransform;
-        public Transform ElementTransform { get; private set; }
-
-        protected MotionControllerInfo controller;
-
         protected virtual void OnAttachToController() { }
         protected virtual void OnDetachFromController() { }
 
-#if UNITY_WSA && UNITY_2017_2_OR_NEWER
-        protected virtual void OnEnable()
+        public override void OnEnable()
         {
             SetChildrenActive(false);
-
-            // Look if the controller has loaded.
-            if (MotionControllerVisualizer.Instance.TryGetControllerModel(handedness, out controller))
-            {
-                AttachElementToController(controller);
-            }
-
-            MotionControllerVisualizer.Instance.OnControllerModelLoaded += AttachElementToController;
-            MotionControllerVisualizer.Instance.OnControllerModelUnloaded += DetachElementFromController;
+            base.OnEnable();
         }
 
-        protected virtual void OnDisable()
+        protected override void AddControllerTransform(MotionControllerInfo newController)
         {
-            if (MotionControllerVisualizer.IsInitialized)
-            {
-                MotionControllerVisualizer.Instance.OnControllerModelLoaded -= AttachElementToController;
-                MotionControllerVisualizer.Instance.OnControllerModelUnloaded -= DetachElementFromController;
-            }
-        }
-
-        protected virtual void OnDestroy()
-        {
-            if (MotionControllerVisualizer.IsInitialized)
+#if UNITY_WSA && UNITY_2017_2_OR_NEWER
+            if (!IsAttached)
             {
-                MotionControllerVisualizer.Instance.OnControllerModelLoaded -= AttachElementToController;
-                MotionControllerVisualizer.Instance.OnControllerModelUnloaded -= DetachElementFromController;
+                base.AddControllerTransform(newController);
```

Comment:
This whole method should be inside the check, right? Not just the `base` call? Since we don't want anything to happen otherwise.

That, and we should likely have a check to make sure `base.AddControllerTransform(newController);` was actually successful, like in the case where the element isn't found and there's nothing to attach to.

Reply:
See previous comment. Fixing this. As far as the check, doesn't *newController.TryGetElement* in the base call provide enough?

Reply:
`newController.TryGetElement` will return out of the base call, but there's nothing here that actually checks to see if that was successful.